RealmBrowser/Classes/RLMRealmBrowserWindowController.m

```diff
@@ -101,6 +108,80 @@ - (IBAction)saveObjcModels:(id)sender
     [RLMModelExporter saveModelsForSchemas:objectSchemas inLanguage:RLMModelExporterLanguageObjectiveC];
 }
 
+- (IBAction)saveCopy:(id)sender
+{
+    NSString *fileName = [self.modelDocument.presentedRealm.realm.path lastPathComponent];
+    NSSavePanel *panel = [NSSavePanel savePanel];
+    [panel setNameFieldStringValue:fileName];
+    [panel beginSheetModalForWindow:self.window completionHandler:^(NSInteger result){
+        if (result != NSFileHandlingPanelOKButton)
+            return;
+        
+        dispatch_async(dispatch_get_main_queue(), ^{
+            NSURL *fileURL = [panel URL];
+            [self exportAndCompactCopyOfRealmFileAtURL:fileURL];
+        });
+    }];
+}
+
+- (void)exportAndCompactCopyOfRealmFileAtURL:(NSURL *)realmFileURL
+{
+    NSError *error = nil;
+    
+    //Check that this won't end up overwriting the original file
+    if ([realmFileURL.path.lowercaseString isEqualToString:self.modelDocument.presentedRealm.realm.path.lowercaseString]) {
+        NSAlert *alert = [[NSAlert alloc] init];
+        alert.messageText = @"You cannot overwrite the original Realm file.";
+        alert.informativeText = @"Please choose a different location in which to save this Realm file.";
+        [alert runModal];
+        return;
+    }
+    
+    //Ensure a file with the same name doesn't already exist
+    BOOL directory = NO;
+    
+    if ([[NSFileManager defaultManager] fileExistsAtPath:realmFileURL.path isDirectory:&directory] && !directory) {
```

Comment:
It does, but I discovered in testing that if you try and save a Realm file over itself while it's open in Browser, it'll throw an exception.

I had a look through Apple's documentation. `NSSavePanel` does *not* allow you to select a directory, it must explicitly be a file (However `NSOpenPanel` apparently does let you select a directory), so this logic should work okay as-is.